BUILD.md

```diff
@@ -0,0 +1,42 @@
+# Building barnabas
+
+Barnabas is currently using `make` as its main build system. Our make build supports several different targets mainly for building and pushing docker images.
+
+<!-- TOC depthFrom:2 -->
+
+- [Docker images](#docker-images)
+    - [Building Docker images](#building-docker-images)
+    - [Tagging and pushing Docker images](#tagging-and-pushing-docker-images)
+- [Building everything](#building-everything)
+- [Release](#release)
+
+<!-- /TOC -->
+
+## Docker images
+
+### Building Docker images
+
+`docker_build` target will build the Docker images provided by the barnabas project. You can build all Barnabas Docker images by calling `make docker_build` from the root of the Barnabas repository. Or you can build an individual Docker image by running `make docker_build` from the subdirectories with their respective dockerfile files - e.g. `kafka_base`, `kafka_statefulsets` etc.
```

Comment:
* "The `docker_build` target..." (add "The")
* "...provided by the Barnabas project." (Barnabas is a proper noun, so has a capital letter).
* "...with their respective `Dockerfile`s..." (dockerfile files is a mouthful!)